### PR TITLE
ecos: init at 2.0.6

### PR DESCRIPTION
**Pull request by bhipple**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bhipple** [comment]
@GrahamcOfBorg build ecos

File: pkgs/development/libraries/science/math/ecos/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  buildPhase = ''
+    make all shared
+  '';
+
+  checkPhase = ''
+    make test
+    ./runecos
+  '';
+
```
Comment (bhipple) [inline pkgs/development/libraries/science/math/ecos/default.nix]: The upstream makefile doesn't ship a proper `make check` or `make install`, but these get the job done.

**Review by kalbasit — APPROVED** [review]

File: pkgs/development/libraries/science/math/ecos/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    cp lib*.a lib*.so $out/lib
+    cp -r include $out/
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A lightweight conic solver for second-order cone programming";
+    homepage = https://www.embotech.com/ECOS;
+    license = licenses.gpl3;
+    maintainers = [ maintainers.bhipple ];
+  };
```
Comment (Infinisil) [inline pkgs/development/libraries/science/math/ecos/default.nix]: Please specify `platforms` here as well

File: pkgs/development/libraries/science/math/ecos/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  version = "2.0.6";
+
+  src = fetchFromGitHub {
+    owner = "embotech";
+    repo = "ecos";
+    rev = "v${version}";
+    sha256 = "11v958j66wq30gxpjpkgl7n3rvla845lygz8fl39pgf1vk9sdyc7";
+  };
+
+  doCheck = true;
```
Comment (Infinisil) [inline pkgs/development/libraries/science/math/ecos/default.nix]: I think this is true by default

**Review by infinisil — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/science/math/ecos/default.nix (hunk 3 above)
Comment (bhipple) [reply pkgs/development/libraries/science/math/ecos/default.nix]: I thought so too, but it's definitely not running the `checkPhase` if I omit this and run `nix-build . -A ecos`.  It's definitely defaults to true in the buildPythonPackage helper, but appears not to be true for the general stdenv?

File: pkgs/development/libraries/science/math/ecos/default.nix (hunk 2 above)
Comment (bhipple) [reply pkgs/development/libraries/science/math/ecos/default.nix]: Added `platforms.all`; this is a simple C library with no dependencies, so I'd expect it to "just work" everywhere.  That said, I've only tested it on `x64_64-linux` (and OfBorg seems to have tested it successfully on `aarch64-linux`).

File: pkgs/development/libraries/science/math/ecos/default.nix (hunk 3 above)
Comment (Infinisil) [reply pkgs/development/libraries/science/math/ecos/default.nix]: Ah yeah you're right, it's false by default

**Review by infinisil — APPROVED** [review]

**Comment by Infinisil** [comment]
Whoops, I should've squashed it

**Pull request merged**